internal/pkg/template/permissions_boundary_integration_test.go

```diff
@@ -0,0 +1,37 @@
+//go:build integration || localintegration
+
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package template
+
+import (
+	"bytes"
+	"github.com/stretchr/testify/require"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+func TestPermissions_Boundary(t *testing.T) {
+	t.Run("every CloudFormation template must contain conditional permissions boundary field for all IAM roles", func(t *testing.T) {
+		cmd := exec.Command("find", "templates", "-name", "*yml")
+		output, err := cmd.Output()
+		require.NoError(t, err, "should return output of 'find' command")
+		
+		files := strings.Fields(string(output))
+		var totalIAMRoles int
+		var totalPermissionsBoundaryFields int
+		for _, file := range files {
+			contents, err := os.ReadFile(file)
+			require.NoError(t, err, "should read file")
+			IAMRoles := bytes.Count(contents, []byte("AWS::IAM::Role"))
+			permissionsBoundaryFields := bytes.Count(contents, []byte("PermissionsBoundary:"))
+				
+			totalIAMRoles += IAMRoles
+			totalPermissionsBoundaryFields += permissionsBoundaryFields
+		}
+		require.True(t, totalIAMRoles == totalPermissionsBoundaryFields, "number of IAM roles does not equal number of permissions boundary fields")
```

Comment:
I think with the current implementation we won't be able to tell which file doesn't have the `PermissionsBoundary:` set.

Can we fail early instead in the for-loop writing exactly which file (the filepath) has the permissions boundary missing? 
For example, try removing one in the templates and see the error message you get from the test to see if that would help the next engineer 😂 

We could even potentially figure out which IAM role doesn't have PermissionsBoundary set by analyzing the line numbers but it's prob over engineering